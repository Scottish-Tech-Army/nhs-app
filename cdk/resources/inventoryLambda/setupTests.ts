// eslint-disable-next-line no-global-assign
console = {
  ...console,
  log: jest.fn(),
  info: jest.fn(),
  debug: jest.fn(),
};

export const TEST_BOXES_TABLE_NAME = "dbtable-boxes";

process.env.BOXES_TABLE_NAME = TEST_BOXES_TABLE_NAME;

beforeEach(() => {
  jest.resetAllMocks();
});
