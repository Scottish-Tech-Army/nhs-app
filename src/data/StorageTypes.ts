export type ItemTemplate = {
  name: string;
  size?: string;
  description: string;
  location: string;
  photo: string;
  quantity?: number;
};

export type BoxTemplate = {
  boxTemplateId: string;
  name: string;
  items: ItemTemplate[];
  count: number;
};

export type StorageAreaTemplate = {
  storageAreaId: string;
  name: string;
  boxes: BoxTemplate[];
};

export type ItemContents = {
  name: string;
  size?: string;
  quantity: number;
};

export type BoxContents = {
  boxTemplateId: string;
  boxNumber: number;
  items: ItemContents[];
};

export type StorageAreaContents = {
  storageAreaId: string;
  boxes: BoxContents[];
};

export type ItemShoppingList = ItemContents;

export type BoxShoppingList = {
  boxTemplateId: string;
  boxNumber: number;
  name: string;
  items: ItemShoppingList[];
};
