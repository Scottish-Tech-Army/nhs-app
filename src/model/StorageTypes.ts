export type ItemTemplate = {
  name: string;
  size?: string;
  description: string;
  location: string;
  photo: string;
  quantity?: number;
};

export type ContainerTemplate = {
  containerTemplateId: string;
  name: string;
  containerType: string;
  items: ItemTemplate[];
  quantity?: number;
};

export type StorageAreaTemplate = {
  storageAreaId: string;
  name: string;
  containers: ContainerTemplate[];
  possibleContainerLocations?: string[];
};

export type StorageAreaGroupTemplate = {
  storageAreaGroupId: string;
  name: string;
  storageAreas: StorageAreaTemplate[];
};

export type MissingContainerItem = {
  name: string;
  size?: string;
  quantity: number;
  location?: string;
};

export type ContainerInputData = {
  checker: string;
  containerTemplateId: string;
  containerNumber?: number;
  storageAreaId: string;
  location?: string;
  name: string;
  missingItems: MissingContainerItem[];
  isFull: boolean;
};

export type ContainerData = ContainerInputData & {
  checkId: string;
  checkTime: string;
};
