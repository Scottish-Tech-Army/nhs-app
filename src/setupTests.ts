// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom";
import localforage from "localforage";
import fetchMock from "jest-fetch-mock";

fetchMock.enableMocks();

jest.mock("localforage");

export const TEST_INVENTORY_API_ENDPOINT = "http://localhost:12345/api/";

process.env.REACT_APP_INVENTORY_API_ENDPOINT = TEST_INVENTORY_API_ENDPOINT;

// eslint-disable-next-line no-global-assign
console = {
  ...console,
  log: jest.fn(),
  info: jest.fn(),
  debug: jest.fn(),
  error: jest.fn(),
};

beforeEach(() => {
  jest.clearAllMocks();
  (localforage.getItem as jest.Mock).mockImplementation(() =>
    Promise.resolve(null)
  );
  (localforage.setItem as jest.Mock).mockImplementation(() =>
    Promise.resolve()
  );
  fetchMock.resetMocks();
});
