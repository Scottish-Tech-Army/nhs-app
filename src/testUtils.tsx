import React from "react";
import { render } from "@testing-library/react";
import type { RenderOptions } from "@testing-library/react";
import { configureStore } from "@reduxjs/toolkit";
import type { PreloadedState } from "@reduxjs/toolkit";
import { Provider } from "react-redux";

import type { AppStore, RootState } from "./data/store";
import boxContentsReducer, { initialState } from "./data/BoxContentsSlice";

interface ExtendedRenderOptions extends Omit<RenderOptions, "queries"> {
  preloadedState?: PreloadedState<RootState>;
  store?: AppStore;
}

export function renderWithProvider(
  component: React.ReactElement,
  {
    preloadedState = { boxContents: initialState },
    store = configureStore({
      reducer: { boxContents: boxContentsReducer },
      preloadedState,
    }),
    ...renderOptions
  }: ExtendedRenderOptions = {}
) {

  return {
    store,
    ...render(<Provider store={store}>{component}</Provider>, renderOptions),
  };
}
