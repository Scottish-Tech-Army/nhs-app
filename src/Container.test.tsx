/* eslint-disable testing-library/no-node-access */
/* eslint-disable testing-library/prefer-screen-queries */
import React from "react";

import { getByRole, screen, waitFor } from "@testing-library/react";
import { Routes, Route } from "react-router-dom";

import { renderWithProvider } from "./testUtils";
import fetchMock from "jest-fetch-mock";

import Container from "./Container";
import { ContainerInputData } from "./model/StorageTypes";
import { TEST_INVENTORY_API_ENDPOINT } from "./setupTests";
import { Auth } from "@aws-amplify/auth";
import { SIGNED_IN } from "./model/auth/AuthStates";

const CHECK_API_ENDPONT = TEST_INVENTORY_API_ENDPOINT + "check";
const TEST_USERNAME = "test user";

jest.mock("@aws-amplify/auth");

describe("Container", () => {
  beforeEach(() => {
    (Auth.currentSession as jest.Mock).mockImplementation(() => {
      return Promise.resolve({
        getIdToken: () => {
          return { getJwtToken: () => "test jwt token" };
        },
      });
    });
  });

  it("rendered a container page", async () => {
    renderWithRoute("trauma-tower", "trauma-chest-drain", "3");
    expect(screen.getByRole("heading")).toHaveTextContent("Trauma Chest Drain - Box 3");

    const inputFields = Array.from(document.querySelectorAll(".display-name"));

    expect(inputFields.map((input) => input.textContent)).toEqual([
      "Blunt dissection chest drainage insertion pack (28Fg)",
      "Sterile gloves (Small)",
      "Sterile gloves (Medium)",
      "Sterile gloves (Large)",
      "Chest drain catheter (28Fr)",
      "Chest drain catheter (32Fr)",
      "Chest drain catheter (36Fr)",
      "ChloraPrep applicator",
      "Lidocaine 1% (5ml / 50mg)",
      "Standard suture pack (Standard)",
      "Mefix roll (5cm x 10m)",
      "Chest drain bottle",
      "Chest drain tubing",
      "Sterile water (H20) bottle (1000ml bottle)",
      "Spencer wells forceps (Straight 20cm)",
    ]);

    expect(screen.getByRole("button", { name: "Save" })).toBeDefined();
  });

  it("does not render if no storageAreaId", async () => {
    const { container } = renderWithRoute("", "trauma-chest-drain", "3");

    expect(container).toHaveTextContent("Unknown path");
  });

  it("does not render if no containerTemplateId", async () => {
    const { container } = renderWithRoute("trauma-tower", "", "3");

    expect(container).toHaveTextContent("Unknown path");
  });

  it("does not render if no containerNumber", async () => {
    const { container } = renderWithRoute("trauma-tower", "trauma-chest-drain", "");

    expect(container).toHaveTextContent("Unknown path");
  });

  it("does not render if unknown storageAreaId", async () => {
    const { container } = renderWithRoute("Unknown", "trauma-chest-drain", "4");

    expect(container.children).toHaveLength(0);

    expect(container).toMatchSnapshot();
  });

  it("does not render if unknown containerTemplateId", async () => {
    const { container } = renderWithRoute("trauma-tower", "Unknown", "4");

    expect(container.children).toHaveLength(0);

    expect(container).toMatchSnapshot();
  });

  it("does not render if unknown containerNumber", async () => {
    const { container } = renderWithRoute("trauma-tower", "trauma-chest-drain", "Unknown");

    expect(container.children).toHaveLength(0);

    expect(container).toMatchSnapshot();
  });

  it("does not render if unknown containerNumber - container number too high", async () => {
    const { container } = renderWithRoute("trauma-tower", "trauma-chest-drain", "7");

    expect(container.children).toHaveLength(0);

    expect(container).toMatchSnapshot();
  });

  it("can save item changes - success", async () => {
    fetchMock.mockResponse("", { status: 200 });

    const { user, history } = renderWithRoute("trauma-tower", "trauma-chest-drain", "3");

    const itemLabel = screen.getByText(
      "Blunt dissection chest drainage insertion pack (28Fg)"
    );

    const increaseButton = getByRole(itemLabel.parentElement!, "button", {
      name: "add item",
    });
    const decreaseButton = getByRole(itemLabel.parentElement!, "button", {
      name: "remove item",
    });
    await user.click(increaseButton);
    await user.click(increaseButton);
    await user.click(increaseButton);
    await user.click(decreaseButton);

    await user.click(screen.getByRole("button", { name: "Save" }));

    const expectedPayload: ContainerInputData = {
      containerTemplateId: "trauma-chest-drain",
      containerNumber: 3,
      storageAreaId: "trauma-tower",
      name: "Trauma Chest Drain",
      missingItems: [
        { quantity: 1, name: "Sterile gloves", size: "Small" },
        { quantity: 1, name: "Sterile gloves", size: "Medium" },
        { quantity: 1, name: "Sterile gloves", size: "Large" },
        { quantity: 1, name: "Chest drain catheter", size: "28Fr" },
        { quantity: 1, name: "Chest drain catheter", size: "32Fr" },
        { quantity: 1, name: "Chest drain catheter", size: "36Fr" },
        { quantity: 2, name: "ChloraPrep applicator" },
        { quantity: 2, name: "Lidocaine 1%", size: "5ml / 50mg" },
        { quantity: 1, name: "Standard suture pack", size: "Standard" },
        { quantity: 1, name: "Mefix roll", size: "5cm x 10m" },
        { quantity: 1, name: "Chest drain bottle" },
        { quantity: 1, name: "Chest drain tubing" },
        {
          quantity: 1,
          name: "Sterile water (H20) bottle",
          size: "1000ml bottle",
        },
        { quantity: 1, name: "Spencer wells forceps", size: "Straight 20cm" },
      ],
      isFull: false,
      checker: TEST_USERNAME,
    };

    await waitFor(() => expect(fetchMock).toHaveBeenCalledTimes(1));
    expect(fetchMock).toHaveBeenCalledWith(
      CHECK_API_ENDPONT,
      expect.objectContaining({
        method: "POST",
        headers: { Authorization: "Bearer test jwt token" },
      })
    );
    const actualPayload = JSON.parse(
      fetchMock.mock.calls[0][1]!.body! as string
    );
    expect(actualPayload).toEqual(expectedPayload);

    expect(history.location.pathname).toEqual("/area/trauma-tower");
  });

  it("can save item changes - failure", async () => {
    fetchMock.mockResponse("", { status: 500 });

    const { user, history } = renderWithRoute("trauma-tower", "trauma-chest-drain", "3");

    const itemLabel = screen.getByText(
      "Blunt dissection chest drainage insertion pack (28Fg)"
    );

    const increaseButton = getByRole(itemLabel.parentElement!, "button", {
      name: "add item",
    });
    const decreaseButton = getByRole(itemLabel.parentElement!, "button", {
      name: "remove item",
    });
    await user.click(increaseButton);
    await user.click(increaseButton);
    await user.click(increaseButton);
    await user.click(decreaseButton);

    await user.click(screen.getByRole("button", { name: "Save" }));

    const expectedPayload: ContainerInputData = {
      containerTemplateId: "trauma-chest-drain",
      containerNumber: 3,
      storageAreaId: "trauma-tower",
      name: "Trauma Chest Drain",
      missingItems: [
        { quantity: 1, name: "Sterile gloves", size: "Small" },
        { quantity: 1, name: "Sterile gloves", size: "Medium" },
        { quantity: 1, name: "Sterile gloves", size: "Large" },
        { quantity: 1, name: "Chest drain catheter", size: "28Fr" },
        { quantity: 1, name: "Chest drain catheter", size: "32Fr" },
        { quantity: 1, name: "Chest drain catheter", size: "36Fr" },
        { quantity: 2, name: "ChloraPrep applicator" },
        { quantity: 2, name: "Lidocaine 1%", size: "5ml / 50mg" },
        { quantity: 1, name: "Standard suture pack", size: "Standard" },
        { quantity: 1, name: "Mefix roll", size: "5cm x 10m" },
        { quantity: 1, name: "Chest drain bottle" },
        { quantity: 1, name: "Chest drain tubing" },
        {
          quantity: 1,
          name: "Sterile water (H20) bottle",
          size: "1000ml bottle",
        },
        { quantity: 1, name: "Spencer wells forceps", size: "Straight 20cm" },
      ],
      isFull: false,
      checker: TEST_USERNAME,
    };

    await waitFor(() => expect(fetchMock).toHaveBeenCalledTimes(1));
    expect(fetchMock).toHaveBeenCalledWith(
      CHECK_API_ENDPONT,
      expect.objectContaining({
        method: "POST",
        headers: { Authorization: "Bearer test jwt token" },
      })
    );
    const actualPayload = JSON.parse(
      fetchMock.mock.calls[0][1]!.body! as string
    );
    expect(actualPayload).toEqual(expectedPayload);

    expect(history.location.pathname).toEqual("/container/trauma-tower/trauma-chest-drain/3");
  });

  it("can mark container as full - success", async () => {
    fetchMock.mockResponse("", { status: 200 });
    const { user, history } = renderWithRoute("trauma-tower", "trauma-chest-drain", "3");

    await user.click(screen.getByRole("button", { name: "FULL" }));

    const expectedPayload: ContainerInputData = {
      containerTemplateId: "trauma-chest-drain",
      containerNumber: 3,
      storageAreaId: "trauma-tower",
      name: "Trauma Chest Drain",
      missingItems: [],
      isFull: true,
      checker: TEST_USERNAME,
    };

    await waitFor(() => expect(fetchMock).toHaveBeenCalledTimes(1));
    expect(fetchMock).toHaveBeenCalledWith(
      CHECK_API_ENDPONT,
      expect.objectContaining({
        method: "POST",
        headers: { Authorization: "Bearer test jwt token" },
      })
    );
    const actualPayload = JSON.parse(
      fetchMock.mock.calls[0][1]!.body! as string
    );
    expect(actualPayload).toEqual(expectedPayload);

    expect(history.location.pathname).toEqual("/area/trauma-tower");
  });

  it("can mark container as full - failure", async () => {
    fetchMock.mockResponse("", { status: 500 });
    const { user, history } = renderWithRoute("trauma-tower", "trauma-chest-drain", "3");

    await user.click(screen.getByRole("button", { name: "FULL" }));

    const expectedPayload: ContainerInputData = {
      containerTemplateId: "trauma-chest-drain",
      containerNumber: 3,
      storageAreaId: "trauma-tower",
      name: "Trauma Chest Drain",
      missingItems: [],
      isFull: true,
      checker: TEST_USERNAME,
    };

    await waitFor(() => expect(fetchMock).toHaveBeenCalledTimes(1));
    expect(fetchMock).toHaveBeenCalledWith(
      CHECK_API_ENDPONT,
      expect.objectContaining({
        method: "POST",
        headers: { Authorization: "Bearer test jwt token" },
      })
    );
    const actualPayload = JSON.parse(
      fetchMock.mock.calls[0][1]!.body! as string
    );
    expect(actualPayload).toEqual(expectedPayload);

    expect(history.location.pathname).toEqual("/container/trauma-tower/trauma-chest-drain/3");
  });

  it("can go to item details", async () => {
    const { user, history } = renderWithRoute("trauma-tower", "trauma-chest-drain", "3");

    const itemLabel = screen.getByText("Chest drain catheter (28Fr)");

    const infoButton = getByRole(itemLabel.parentElement!, "button", {
      name: "item information",
    });
    await user.click(infoButton);

    expect(history.location.pathname).toEqual("/item/trauma-chest-drain/4");
  });

  it("can return to previous page", async () => {
    const { user, history } = renderWithRoute("trauma-tower", "trauma-chest-drain", "3");

    await user.click(screen.getByRole("button", { name: "back" }));

    expect(history.location.pathname).toEqual("/");
  });

  it("renders correctly", () => {
    const { container } = renderWithRoute("trauma-tower", "trauma-chest-drain", "3");

    expect(container).toMatchSnapshot();
  });

  function renderWithRoute(storageAreaId: string, containerTemplateId: string, containerNumber: string) {
    // Add routes to get the contents of useParams populated
    return renderWithProvider(
      <Routes>
        <Route path="container/:storageAreaId/:containerTemplateId/:containerNumber" element={<Container />} />
        <Route path="*" element={<div>Unknown path</div>} />
      </Routes>,
      {
        initialRoutes: ["/", `/container/${storageAreaId}/${containerTemplateId}/${containerNumber}`],
        preloadedState: {
          auth: {
            authState: SIGNED_IN,
            user: { name: TEST_USERNAME },
            errorMessage: "",
          },
        },
      }
    );
  }
});
