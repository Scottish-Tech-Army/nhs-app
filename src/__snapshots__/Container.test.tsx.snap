// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Multiple Instance Container does not render if unknown containerNumber - container number too high 1`] = `<div />`;

exports[`Multiple Instance Container does not render if unknown containerNumber 1`] = `<div />`;

exports[`Multiple Instance Container does not render if unknown containerTemplateId 1`] = `<div />`;

exports[`Multiple Instance Container does not render if unknown storageAreaId 1`] = `<div />`;

exports[`Multiple Instance Container renders correctly 1`] = `
<div>
  <div
    class="container-details with-location"
  >
    <header>
      <h1>
        Trauma Tower
        <br />
        Trauma Chest Drain - BoxÂ 3
      </h1>
      <button
        aria-label="back"
        class="back"
        type="button"
      >
        <svg>
          arrow-left.svg
        </svg>
      </button>
      <button
        class="full"
        type="button"
      >
        FULL
      </button>
    </header>
    <main>
      <div
        class="scroll"
      >
        <div
          class="item"
        >
          <div
            class="display-name"
          >
            Blunt dissection chest drainage insertion pack (28Fg)
          </div>
          <div
            class="controls"
          >
            <button
              aria-label="item information"
              class="item-info"
              type="button"
            >
              <svg>
                info-circle.svg
              </svg>
            </button>
            <div
              class="quantity"
            >
              <span
                class="actual-quantity-bad"
              >
                0
              </span>
              <span>
                 / 
              </span>
              <span
                class="expected-quantity"
              >
                1
              </span>
            </div>
            <button
              aria-label="remove item"
              class="change-quantity"
              disabled=""
              type="button"
            >
              <svg>
                minus.svg
              </svg>
            </button>
            <button
              aria-label="add item"
              class="change-quantity"
              type="button"
            >
              <svg>
                add.svg
              </svg>
            </button>
          </div>
        </div>
        <div
          class="item"
        >
          <div
            class="display-name"
          >
            Sterile gloves (Small)
          </div>
          <div
            class="controls"
          >
            <button
              aria-label="item information"
              class="item-info"
              type="button"
            >
              <svg>
                info-circle.svg
              </svg>
            </button>
            <div
              class="quantity"
            >
              <span
                class="actual-quantity-bad"
              >
                0
              </span>
              <span>
                 / 
              </span>
              <span
                class="expected-quantity"
              >
                1
              </span>
            </div>
            <button
              aria-label="remove item"
              class="change-quantity"
              disabled=""
              type="button"
            >
              <svg>
                minus.svg
              </svg>
            </button>
            <button
              aria-label="add item"
              class="change-quantity"
              type="button"
            >
              <svg>
                add.svg
              </svg>
            </button>
          </div>
        </div>
        <div
          class="item"
        >
          <div
            class="display-name"
          >
            Sterile gloves (Medium)
          </div>
          <div
            class="controls"
          >
            <button
              aria-label="item information"
              class="item-info"
              type="button"
            >
              <svg>
                info-circle.svg
              </svg>
            </button>
            <div
              class="quantity"
            >
              <span
                class="actual-quantity-bad"
              >
                0
              </span>
              <span>
                 / 
              </span>
              <span
                class="expected-quantity"
              >
                1
              </span>
            </div>
            <button
              aria-label="remove item"
              class="change-quantity"
              disabled=""
              type="button"
            >
              <svg>
                minus.svg
              </svg>
            </button>
            <button
              aria-label="add item"
              class="change-quantity"
              type="button"
            >
              <svg>
                add.svg
              </svg>
            </button>
          </div>
        </div>
        <div
          class="item"
        >
          <div
            class="display-name"
          >
            Sterile gloves (Large)
          </div>
          <div
            class="controls"
          >
            <button
              aria-label="item information"
              class="item-info"
              type="button"
            >
              <svg>
                info-circle.svg
              </svg>
            </button>
            <div
              class="quantity"
            >
              <span
                class="actual-quantity-bad"
              >
                0
              </span>
              <span>
                 / 
              </span>
              <span
                class="expected-quantity"
              >
                1
              </span>
            </div>
            <button
              aria-label="remove item"
              class="change-quantity"
              disabled=""
              type="button"
            >
              <svg>
                minus.svg
              </svg>
            </button>
            <button
              aria-label="add item"
              class="change-quantity"
              type="button"
            >
              <svg>
                add.svg
              </svg>
            </button>
          </div>
        </div>
        <div
          class="item"
        >
          <div
            class="display-name"
          >
            Chest drain catheter (28Fr)
          </div>
          <div
            class="controls"
          >
            <button
              aria-label="item information"
              class="item-info"
              type="button"
            >
              <svg>
                info-circle.svg
              </svg>
            </button>
            <div
              class="quantity"
            >
              <span
                class="actual-quantity-bad"
              >
                0
              </span>
              <span>
                 / 
              </span>
              <span
                class="expected-quantity"
              >
                1
              </span>
            </div>
            <button
              aria-label="remove item"
              class="change-quantity"
              disabled=""
              type="button"
            >
              <svg>
                minus.svg
              </svg>
            </button>
            <button
              aria-label="add item"
              class="change-quantity"
              type="button"
            >
              <svg>
                add.svg
              </svg>
            </button>
          </div>
        </div>
        <div
          class="item"
        >
          <div
            class="display-name"
          >
            Chest drain catheter (32Fr)
          </div>
          <div
            class="controls"
          >
            <button
              aria-label="item information"
              class="item-info"
              type="button"
            >
              <svg>
                info-circle.svg
              </svg>
            </button>
            <div
              class="quantity"
            >
              <span
                class="actual-quantity-bad"
              >
                0
              </span>
              <span>
                 / 
              </span>
              <span
                class="expected-quantity"
              >
                1
              </span>
            </div>
            <button
              aria-label="remove item"
              class="change-quantity"
              disabled=""
              type="button"
            >
              <svg>
                minus.svg
              </svg>
            </button>
            <button
              aria-label="add item"
              class="change-quantity"
              type="button"
            >
              <svg>
                add.svg
              </svg>
            </button>
          </div>
        </div>
        <div
          class="item"
        >
          <div
            class="display-name"
          >
            Chest drain catheter (36Fr)
          </div>
          <div
            class="controls"
          >
            <button
              aria-label="item information"
              class="item-info"
              type="button"
            >
              <svg>
                info-circle.svg
              </svg>
            </button>
            <div
              class="quantity"
            >
              <span
                class="actual-quantity-bad"
              >
                0
              </span>
              <span>
                 / 
              </span>
              <span
                class="expected-quantity"
              >
                1
              </span>
            </div>
            <button
              aria-label="remove item"
              class="change-quantity"
              disabled=""
              type="button"
            >
              <svg>
                minus.svg
              </svg>
            </button>
            <button
              aria-label="add item"
              class="change-quantity"
              type="button"
            >
              <svg>
                add.svg
              </svg>
            </button>
          </div>
        </div>
        <div
          class="item"
        >
          <div
            class="display-name"
          >
            ChloraPrep applicator (3ml)
          </div>
          <div
            class="controls"
          >
            <button
              aria-label="item information"
              class="item-info"
              type="button"
            >
              <svg>
                info-circle.svg
              </svg>
            </button>
            <div
              class="quantity"
            >
              <span
                class="actual-quantity-bad"
              >
                0
              </span>
              <span>
                 / 
              </span>
              <span
                class="expected-quantity"
              >
                2
              </span>
            </div>
            <button
              aria-label="remove item"
              class="change-quantity"
              disabled=""
              type="button"
            >
              <svg>
                minus.svg
              </svg>
            </button>
            <button
              aria-label="add item"
              class="change-quantity"
              type="button"
            >
              <svg>
                add.svg
              </svg>
            </button>
          </div>
        </div>
        <div
          class="item"
        >
          <div
            class="display-name"
          >
            Lidocaine 1% vial (5ml / 50mg)
          </div>
          <div
            class="controls"
          >
            <button
              aria-label="item information"
              class="item-info"
              type="button"
            >
              <svg>
                info-circle.svg
              </svg>
            </button>
            <div
              class="quantity"
            >
              <span
                class="actual-quantity-bad"
              >
                0
              </span>
              <span>
                 / 
              </span>
              <span
                class="expected-quantity"
              >
                2
              </span>
            </div>
            <button
              aria-label="remove item"
              class="change-quantity"
              disabled=""
              type="button"
            >
              <svg>
                minus.svg
              </svg>
            </button>
            <button
              aria-label="add item"
              class="change-quantity"
              type="button"
            >
              <svg>
                add.svg
              </svg>
            </button>
          </div>
        </div>
        <div
          class="item"
        >
          <div
            class="display-name"
          >
            Suture pack (Standard)
          </div>
          <div
            class="controls"
          >
            <button
              aria-label="item information"
              class="item-info"
              type="button"
            >
              <svg>
                info-circle.svg
              </svg>
            </button>
            <div
              class="quantity"
            >
              <span
                class="actual-quantity-bad"
              >
                0
              </span>
              <span>
                 / 
              </span>
              <span
                class="expected-quantity"
              >
                1
              </span>
            </div>
            <button
              aria-label="remove item"
              class="change-quantity"
              disabled=""
              type="button"
            >
              <svg>
                minus.svg
              </svg>
            </button>
            <button
              aria-label="add item"
              class="change-quantity"
              type="button"
            >
              <svg>
                add.svg
              </svg>
            </button>
          </div>
        </div>
        <div
          class="item"
        >
          <div
            class="display-name"
          >
            Mefix roll (5cm x 10m)
          </div>
          <div
            class="controls"
          >
            <button
              aria-label="item information"
              class="item-info"
              type="button"
            >
              <svg>
                info-circle.svg
              </svg>
            </button>
            <div
              class="quantity"
            >
              <span
                class="actual-quantity-bad"
              >
                0
              </span>
              <span>
                 / 
              </span>
              <span
                class="expected-quantity"
              >
                1
              </span>
            </div>
            <button
              aria-label="remove item"
              class="change-quantity"
              disabled=""
              type="button"
            >
              <svg>
                minus.svg
              </svg>
            </button>
            <button
              aria-label="add item"
              class="change-quantity"
              type="button"
            >
              <svg>
                add.svg
              </svg>
            </button>
          </div>
        </div>
        <div
          class="item"
        >
          <div
            class="display-name"
          >
            Chest drain bottle
          </div>
          <div
            class="controls"
          >
            <button
              aria-label="item information"
              class="item-info"
              type="button"
            >
              <svg>
                info-circle.svg
              </svg>
            </button>
            <div
              class="quantity"
            >
              <span
                class="actual-quantity-bad"
              >
                0
              </span>
              <span>
                 / 
              </span>
              <span
                class="expected-quantity"
              >
                1
              </span>
            </div>
            <button
              aria-label="remove item"
              class="change-quantity"
              disabled=""
              type="button"
            >
              <svg>
                minus.svg
              </svg>
            </button>
            <button
              aria-label="add item"
              class="change-quantity"
              type="button"
            >
              <svg>
                add.svg
              </svg>
            </button>
          </div>
        </div>
        <div
          class="item"
        >
          <div
            class="display-name"
          >
            Chest drain tubing
          </div>
          <div
            class="controls"
          >
            <button
              aria-label="item information"
              class="item-info"
              type="button"
            >
              <svg>
                info-circle.svg
              </svg>
            </button>
            <div
              class="quantity"
            >
              <span
                class="actual-quantity-bad"
              >
                0
              </span>
              <span>
                 / 
              </span>
              <span
                class="expected-quantity"
              >
                1
              </span>
            </div>
            <button
              aria-label="remove item"
              class="change-quantity"
              disabled=""
              type="button"
            >
              <svg>
                minus.svg
              </svg>
            </button>
            <button
              aria-label="add item"
              class="change-quantity"
              type="button"
            >
              <svg>
                add.svg
              </svg>
            </button>
          </div>
        </div>
        <div
          class="item"
        >
          <div
            class="display-name"
          >
            Sterile water (H20) bottle (1000ml bottle)
          </div>
          <div
            class="controls"
          >
            <button
              aria-label="item information"
              class="item-info"
              type="button"
            >
              <svg>
                info-circle.svg
              </svg>
            </button>
            <div
              class="quantity"
            >
              <span
                class="actual-quantity-bad"
              >
                0
              </span>
              <span>
                 / 
              </span>
              <span
                class="expected-quantity"
              >
                1
              </span>
            </div>
            <button
              aria-label="remove item"
              class="change-quantity"
              disabled=""
              type="button"
            >
              <svg>
                minus.svg
              </svg>
            </button>
            <button
              aria-label="add item"
              class="change-quantity"
              type="button"
            >
              <svg>
                add.svg
              </svg>
            </button>
          </div>
        </div>
        <div
          class="item"
        >
          <div
            class="display-name"
          >
            Spencer wells forceps (Straight 20cm)
          </div>
          <div
            class="controls"
          >
            <button
              aria-label="item information"
              class="item-info"
              type="button"
            >
              <svg>
                info-circle.svg
              </svg>
            </button>
            <div
              class="quantity"
            >
              <span
                class="actual-quantity-bad"
              >
                0
              </span>
              <span>
                 / 
              </span>
              <span
                class="expected-quantity"
              >
                1
              </span>
            </div>
            <button
              aria-label="remove item"
              class="change-quantity"
              disabled=""
              type="button"
            >
              <svg>
                minus.svg
              </svg>
            </button>
            <button
              aria-label="add item"
              class="change-quantity"
              type="button"
            >
              <svg>
                add.svg
              </svg>
            </button>
          </div>
        </div>
      </div>
    </main>
    <footer>
      <div
        class="container-locations"
      >
        <label
          for="location"
        >
          Location
        </label>
        <select
          id="location"
          name="location"
        >
          <option
            value="Resus Stock Cupboard"
          >
            Resus Stock Cupboard
          </option>
          <option
            value="Resus 1a"
          >
            Resus 1a
          </option>
          <option
            value="Resus 1b"
          >
            Resus 1b
          </option>
          <option
            value="Resus 1c"
          >
            Resus 1c
          </option>
          <option
            value="Resus 2"
          >
            Resus 2
          </option>
          <option
            value="Ward Clerk's Desk"
          >
            Ward Clerk's Desk
          </option>
        </select>
      </div>
      <button
        class="save"
        type="button"
      >
        Save
      </button>
    </footer>
  </div>
</div>
`;

exports[`Single Instance Container renders correctly 1`] = `
<div>
  <div
    class="container-details "
  >
    <header>
      <h1>
        Airway Trolley 2
        <br />
        Side
      </h1>
      <button
        aria-label="back"
        class="back"
        type="button"
      >
        <svg>
          arrow-left.svg
        </svg>
      </button>
      <button
        class="full"
        type="button"
      >
        FULL
      </button>
    </header>
    <main>
      <div
        class="scroll"
      >
        <div
          class="item"
        >
          <div
            class="display-name"
          >
            Aintree Catheter
          </div>
          <div
            class="controls"
          >
            <button
              aria-label="item information"
              class="item-info"
              type="button"
            >
              <svg>
                info-circle.svg
              </svg>
            </button>
            <div
              class="quantity"
            >
              <span
                class="actual-quantity-bad"
              >
                0
              </span>
              <span>
                 / 
              </span>
              <span
                class="expected-quantity"
              >
                1
              </span>
            </div>
            <button
              aria-label="remove item"
              class="change-quantity"
              disabled=""
              type="button"
            >
              <svg>
                minus.svg
              </svg>
            </button>
            <button
              aria-label="add item"
              class="change-quantity"
              type="button"
            >
              <svg>
                add.svg
              </svg>
            </button>
          </div>
        </div>
        <div
          class="item"
        >
          <div
            class="display-name"
          >
            Bougie (Cook Frova)
          </div>
          <div
            class="controls"
          >
            <button
              aria-label="item information"
              class="item-info"
              type="button"
            >
              <svg>
                info-circle.svg
              </svg>
            </button>
            <div
              class="quantity"
            >
              <span
                class="actual-quantity-bad"
              >
                0
              </span>
              <span>
                 / 
              </span>
              <span
                class="expected-quantity"
              >
                2
              </span>
            </div>
            <button
              aria-label="remove item"
              class="change-quantity"
              disabled=""
              type="button"
            >
              <svg>
                minus.svg
              </svg>
            </button>
            <button
              aria-label="add item"
              class="change-quantity"
              type="button"
            >
              <svg>
                add.svg
              </svg>
            </button>
          </div>
        </div>
      </div>
    </main>
    <footer>
      <button
        class="save"
        type="button"
      >
        Save
      </button>
    </footer>
  </div>
</div>
`;
